packages/matter.js/src/device/Endpoint.ts

```diff
@@ -21,28 +17,41 @@ import { FixedLabelCluster, UserLabelCluster } from "../cluster/LabelCluster.js"
 import { ClusterClientObj } from "../cluster/client/ClusterClient.js";
 import { ClusterServerObj, ClusterServerObjForCluster } from "../cluster/server/ClusterServer.js";
 import { InteractionClient } from "../protocol/interaction/InteractionClient.js";
-import { AllClustersMap } from "../cluster/index.js";
+import { AllClustersMap, BridgedDeviceBasicInformationCluster } from "../cluster/index.js";
+import { BasicInformationCluster } from "../cluster/BasicInformationCluster.js";
+
+export interface EndpointOptions {
+    endpointId?: number;
+    uniqueId?: string;
```

Comment:
endpointId is also unique I think?  What about something like "persistentId" or "deviceIdentificationKey"?

Reply:
In fact with this you explicitely set the endpoint-ID (aka id attrobute of the endpoint). If you provide duplicates there is no correction or such  and you get an error, so it really is the endpoint ID. if you provide this this is also not stored in any way. So name intended

Reply:
Oh sorry wasn't clear, I meant instead of "uniqueID"

Reply:
uniqueEndpointKey?
uniqueEndpointStorageKey?

Reply:
Yeah those work, the "endpoint" part might be implied by fact it's in endpoint options but uniqueStorageKey or uniqueEndpointStorageKey seem fine

Reply:
kk

Reply:
I just realized that you were using the terminology from BasicInformationCluster.  So maybe your original uniqueId is the better choice if it maps directly into BasicInformation.UniqueID.  Maybe a /** comment would be helpful

Reply:
All good the renaming mames sense because it is "Just a custom such unique id" ...so we can also rename it